dashboard/src/app/workspace/share-workspace/share-workspace.controller.js

```diff
@@ -0,0 +1,266 @@
+   * @returns {*} promise with store permissions request
+   */
+  storeRecipePermissions(user) {
+    let permission = {};
+    permission.user = user.id;
+    permission.domain = this.recipeDomain;
+    permission.instance = this.recipeId;
+    permission.actions = ['read'];
+
+    return this.codenvyPermissions.storePermissions(permission);
+  }
+
+  /**
+   * Removes user permissions for current workspace
+   *
+   * @param user user
+   */
+  removePermissions(user) {
+    this.isLoading = true;
+    this.codenvyPermissions.removeWorkspacePermissions(user.permissions.instance, user.id).then(() => {
+      this.refreshWorkspacePermissions();
+    }, (error) => {
+      this.isLoading = false;
+      this.cheNotification.showError(error.data && error.data.message ? error.data.message : 'Failed to remove user ' + user.email + ' permissions.');
+    });
+  }
+
+  /**
+   * Find user by its email.
+   *
+   * @param email user's email
+   * @returns {*}
+   */
+  findUser(email) {
+    //Prevents mensioning same users twice:
+    if (this.existingUsers.has(email)) {
+      return null;
+    }
+
+    //Displays user name instead of email
+    if (this.codenvyUser.getUserByAlias(email)) {
+      let user = this.codenvyUser.getUserByAlias(email);
+      this.existingUsers.set(email, user);
+      return email;
+    }
+
+    let findUser = this.codenvyUser.fetchUserByAlias(email).then(() => {
+      let user = this.codenvyUser.getUserByAlias(email);
+      this.existingUsers.set(email, user);
+    }, (error) => {
+      this.cheNotification.showError(error.data && error.data.message ? error.data.message : 'Failed to find user ' + email + '.');
+    });
+
+    return email;
+  }
+
+  /**
+   * Removes removed email from the list of registered users.
+   *
+   * @param email email to remove
+   */
+  onRemoveEmail(email) {
+    this.existingUsers.delete(email);
+  }
+
+  /**
+   * Returns user's name by email
+   *
+   * @param email user email
+   * @returns {user.name|*} user's name
+   */
+  getUserName(email) {
+    let user = this.codenvyUser.getUserByAlias(email);
+    return user ? user.name : email;
+  }
+
+  /**
+   * Checks user exists in the system (is reqistered).
+   *
+   * @param email user's email to be checked
+   * @returns {boolean}
+   */
+  isUserExists(email) {
+    return (this.codenvyUser.getUserByAlias(email) !== undefined);
+  }
+
+  /**
+   * Detects recipe id from it's location.
+   *
+   * (Format of the url : http://{host}/api/recipe/{recipe id}/script).
+   *
+   * @returns {*|string}
+   */
+  getRecipeId() {
+    let workspace = this.cheWorkspace.getWorkspaceById(this.workspaceId);
+    let environments = workspace.config.environments;
+    let defaultEnvName = workspace.config.defaultEnv;
+
+    let defaultEnvironment = this.lodash.find(environments, (environment) => {
+        return environment.name === defaultEnvName;
+    });
+
+    let recipeLocation = defaultEnvironment.machineConfigs[0].source.location;
```

Comment:
maybe now we've to use loadash as well to get the machine which is "dev" machine